.typespec/chat/models.tsp

```diff
@@ -182,38 +182,14 @@ model CreateChatCompletionRequest {
 }
 
 /** Represents a chat completion response returned by model, based on the provided input. */
-model CreateChatCompletionResponse {
+model CreateChatCompletionResponse<T extends ChatCompletionChoice = ChatCompletionChoice> {
```

Comment:
This is a tool-generated TypeSpec file and we should figure out how we want to approach that problem:

- The default answer is that we could put the whole type into `custom.tsp` and add an ingestion rule for it; this would unblock using the templated type but unfortunately require separating quite a lot from the rule-driven basis.
- The harder (but perhaps tenable?) approach would be to define an OpenAPI-based rule that generalizes this templating; looking at `CreateChatCompletionResponse`, I'd envision it needing to be something like "if a schema type has a single array of objects or `oneOf` references..."
  - Extract the array item definition into its own type that concatenates the parent name and singular field name, e.g. here `CreateChatCompletionResponseChoice`
  - Update the containing type to be templated to e.g. `<TChoice extends CreateChatCompletionResponseChoice>`
  - Use the templated type, e.g. `CreateChatCompletionResponse<CreateChatCompletionResponseChoice>`, wherever the untemplated type was previously used

That seems a bit messy, but perhaps attainable.

Reply:
Oh I wasn't aware these bits where generated. 

I see what you mean, this is going to be painful to work around. I have another idea that I will explore and probably have an answer for you before we talk today, but it involves using [OmitProperties](https://typespec.io/docs/standard-library/built-in-data-types#OmitProperties) as a model instead of a decorator. It would seem possible to have something like 

```ts
alias StrippedChatCompletionResponse = OmittedProperties<OpenAI.ChatCompletionReponse, "stream_options" | "choices">;

model AzureChatCompletionResponse {
  ...StrippedChatCompletionResponse;

  choices: AzureChatCompletionChoice[];
}
```

I sort of like this approach, but we lose the inheritance relationship. I added `"stream_options"` too just to exemplify that we can remove several properties at once.

But another consideration, I gather from reading the TSP docs and the sort of comments we've gotten on meetings, I becoming increasingly convinced that there is not much attached semantically to an `extends` relationship between models. At least in this point in time in TSP, it's just another means of composing members in models.

Alternatively, I could inline the type above 🤔 like so:

```ts
model AzureChatCompletionsResponse extends OmittedProperties<OponAI.ChatCompletionResponse, "stream_response" | "choices"> {

  choices: AzureChatCompletionsChoice[];
}
```

Ok, I think I will take this to VSCode instead 😅

Reply:
Ok, it seems that I have partial good news. The 2nd approach works, where we extend a version of the model that has the omitted field. At the swagger level though, there is nothing signaling the inheritance relationship, like we can see is still the case in `main` [here[(https://github.com/joseharriaga/openai-in-typespec/blob/9ce03fe51781928f68522cf3a556fb846159a2e3/.openapi3.azure/openapi3-azure-openai.yaml#L62-L74)

In a certain sense I think this makes sense, stripping a model of a member renders it a different model anyway and as you can see in the swagger of this PR, we get an `allOf` with the remaining fields. 

Between this and the template type version, I am not entirely sure which one is the clearest. Also not sure for what artifact we want to optimize clarity.

Reply:
One thing I forgot to mention: we still would need to have `OpenAI.ChatCompletionChoice` as a named model, unless we are ok, with in-lining the fields directly into its Azure counterpart manually.